great_ape_safe/ape_api/ebtc.py

```diff
@@ -793,12 +846,96 @@ def priceFeed_set_fallback_caller(
                 timelock, target.address, 0, data, EmptyBytes32, salt, delay + 1
             )
 
+    #### ===== EBTC FEED ===== ####
+
+    def ebtcFeed_set_primary_oracle(self, address, salt=EmptyBytes32):
+        """
+        @dev Sets the primary Oracle on the EBTC Feed.
+        @param address The address of the new primary Oracle
+        @param salt Value used to generate a unique ID for a transaction with identical parameters than an existing.
+        """
+        ## Check if tx is already scheduled
+        target = self.ebtc_feed
+        data = target.setPrimaryOracle.encode_input(address)
+        id = self.highsec_timelock.hashOperation(
+            target.address, 0, data, EmptyBytes32, salt
+        )
+
+        if self.highsec_timelock.isOperation(id):
+            self.execute_timelock(
+                self.highsec_timelock,
+                target.address,
+                0,
+                data,
+                EmptyBytes32,
+                salt,
+            )
+            assert (
+                self.ebtc_feed.primaryOracle() == address
+            ), "Error: Primary Oracle not set"
+        else:
+            delay = self.highsec_timelock.getMinDelay()
+            self.schedule_timelock(
+                self.highsec_timelock,
+                target.address,
+                0,
+                data,
+                EmptyBytes32,
+                salt,
+                delay + 1,
+            )
+
+    def ebtcFeed_set_secondary_oracle(
+        self, address, salt=EmptyBytes32, use_high_sec=False
+    ):
+        """
+        @dev Sets the secondary Oracle on the EBTC Feed.
+        @param address The address of the new secondary Oracle
+        @param salt Value used to generate a unique ID for a transaction with identical parameters than an existing.
+        @param use_high_sec If true, use the high security timelock. Otherwise, use the low security timelock.
+        """
+
+        if use_high_sec:
+            timelock = self.highsec_timelock
+        else:
+            timelock = self.lowsec_timelock
+
+        ## Check if tx is already scheduled
+        target = self.ebtc_feed
+        data = target.setSecondaryOracle.encode_input(address)
+        id = timelock.hashOperation(target.address, 0, data, EmptyBytes32, salt)
+
+        if timelock.isOperation(id):
+            self.execute_timelock(
```

Comment:
for this boilerplate code, maybe could use a common internal function like:

```
executeOrDelayTask(use_high_sec, target, data, salt, someAssertFuncAfter)
```

Reply:
Good call, addressed: https://github.com/ebtc-protocol/ebtc-multisig/pull/17/commits/42c64d1ee9b4914588221722132ba035da5085a9